core/services/pipeline/task.bridge_test.go

```diff
@@ -1015,3 +1050,94 @@ func TestBridgeTask_Headers(t *testing.T) {
 		assert.Equal(t, []string{"Content-Length", "38", "Content-Type", "footype", "User-Agent", "Go-http-client/1.1", "X-Header-1", "foo", "X-Header-2", "bar"}, allHeaders(headers))
 	})
 }
+
+func TestBridgeTask_AdapterResponseStatusFailure(t *testing.T) {
```

Comment:
Could you also add a test that tests the `bestEffortExtractEAStatus` method directly for all of its cases?